specification/advisor/resource-manager/Microsoft.Advisor/preview/2022-02-01-preview/predictRecommendation.json

```diff
@@ -0,0 +1,348 @@
+        },
+        "x-ms-examples": {
+          "OperationsList": {
+            "$ref": "./examples/OperationsList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Advisor/predict": {
+      "post": {
+        "tags": [
+          "Prediction"
+        ],
+        "summary": "Predicts a recommendation.",
+        "operationId": "Predict",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "name": "predictionRequest",
+            "in": "body",
+            "description": "Parameters for predict recommendation.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PredictionRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully obtained prediction.",
+            "schema": {
+              "$ref": "#/definitions/PredictionResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ArmErrorResponse"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-examples": {
+          "Predict": {
+            "$ref": "./examples/Predict.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ArmErrorResponse": {
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/ARMErrorResponseBody"
+        }
+      }
+    },
+    "ARMErrorResponseBody": {
+      "description": "ARM error response body.",
+      "type": "object",
+      "properties": {
+        "message": {
+          "description": "Gets or sets the string that describes the error in detail and provides debugging information.",
+          "type": "string"
+        },
+        "code": {
+          "description": "Gets or sets the string that can be used to programmatically identify the error.",
+          "type": "string"
+        }
+      }
+    },
+    "ShortDescription": {
+      "description": "A summary of the recommendation.",
+      "type": "object",
+      "properties": {
+        "solution": {
+          "description": "The remediation action suggested by the recommendation.",
+          "type": "string"
+        }
+      }
+    },
+    "OperationEntityListResult": {
+      "description": "The list of Advisor operations.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The link used to get the next page of operations.",
+          "type": "string"
+        },
+        "value": {
+          "description": "The list of operations.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OperationEntity"
```

Comment:
>OperationEntity [](http://example.com/codeflow?start=35&length=15)

Can you please use the common definitions for these ?

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json#L363](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json#L363)

Reply:
I updated to use the common definition.